typescript/web/src/components/import-button/import-images-modal/modal-dropzone/import-dropped-files/import-datasets.tsx

```diff
@@ -0,0 +1,95 @@
+import { ApolloClient, gql } from "@apollo/client";
+import { ExportFormat } from "@labelflow/graphql-types";
+import { v4 as uuidv4 } from "uuid";
+import mime from "mime-types";
+
+import Bluebird from "bluebird";
+import { uploadFile } from "../../../../../utils/upload-file";
+import { DroppedFile, SetUploadStatuses } from "../../types";
+
+import { CONCURRENCY } from "../../constants";
+
+const importDatasetMutation = gql`
+  mutation importDataset(
+    $where: DatasetWhereUniqueInput!
+    $data: DatasetImportInput!
+  ) {
+    importDataset(where: $where, data: $data) {
+      error
+    }
+  }
+`;
+const importDataset = async ({
+  apolloClient,
+  datasetId,
+  workspaceId,
+  file,
+}: {
+  apolloClient: ApolloClient<any>;
+  datasetId: string;
+  workspaceId: string;
+  file: Blob;
+}) => {
+  const key = `${workspaceId}/${datasetId}/import-datasets/${uuidv4()}-${new Date().toISOString()}.${mime.extension(
+    file.type
+  )}`;
+  const url = await uploadFile({ file, key, apolloClient });
+
+  const dataImportDataset = await apolloClient.mutate({
+    mutation: importDatasetMutation,
+    variables: {
+      where: { id: datasetId },
+      data: {
+        url,
+        format: ExportFormat.Coco,
+        options: {
+          coco: {
+            annotationsOnly: file.type === "application/json",
+          },
+        },
+      },
+    },
+    refetchQueries: [
+      "getDatasetData",
+      "getImageLabels",
+      "getLabel",
+      "countLabelsOfDataset",
+      "getDatasetLabelClasses",
```

Comment:
While it's necessary to tell which queries are invalidated, I'm afraid we'll have a lot of bugs when forgetting to add new queries in such arrays. Isn't there a way to make it more future-proof?

Reply:
That's not something I introduced in this PR. 
Changing this would be quite difficult. It implies to rely only on cache updates. 

At some point we discussed migrating from refetch queries to cache updates (in https://github.com/labelflow/labelflow/issues/366).